goal_src/levels/rolling/rolling-obs.gc

```diff
@@ -5,3 +5,1779 @@
+  :longest-edge (meters 0)
+  )
+
+(deftype pusher-base (process-drawable)
+  ((root-override collide-shape-moving  :offset        112)
+   (max-frame     float                 :offset-assert 176)
+   )
+  :heap-base #x50
+  :method-count-assert 20
+  :size-assert         #xb4
+  :flag-assert         #x14005000b4
+  )
+
+
+(deftype pusher (pusher-base)
+  ((sync sync-info-paused :inline :offset-assert 180)
+   (cyl  cylinder         :inline :offset-assert 208)
+   )
+  :heap-base #x90
+  :method-count-assert 20
+  :size-assert         #xf8
+  :flag-assert         #x14009000f8
+  (:states
+    pusher-idle
+    )
+  )
+
+
+(deftype gorge-pusher (pusher-base)
+  ((min-frame float  :offset-assert 180)
+   )
+  :heap-base #x50
+  :method-count-assert 20
+  :size-assert         #xb8
+  :flag-assert         #x14005000b8
+  (:states
+    gorge-pusher-idle
+    )
+  )
+
+
+(defskelgroup *pusher-sg* pusher
+  0
+  2
+  ((1 (meters 999999)))
+  :bounds (static-spherem 0 0 0 10)
+  :longest-edge (meters 0)
+  )
+
+(defbehavior pusher-base-init pusher-base ()
+  (set! (-> self mask) (logior (process-mask enemy platform) (-> self mask)))
+  (let ((gp-0 (new 'process 'collide-shape-moving self (collide-list-enum hit-by-others))))
+    (set! (-> gp-0 dynam) (copy *standard-dynamics* 'process))
+    (set! (-> gp-0 reaction) default-collision-reaction)
+    (set! (-> gp-0 no-reaction)
+          (the-as (function collide-shape-moving collide-shape-intersect vector vector none) nothing)
+          )
+    (let ((s5-0 (new 'process 'collide-shape-prim-group gp-0 (the-as uint 1) 0)))
+      (set! (-> s5-0 prim-core collide-as) (the-as uint 2048))
+      (set! (-> s5-0 collide-with) (the-as uint 16))
+      (set! (-> s5-0 prim-core action) (the-as uint 3))
+      (set! (-> s5-0 transform-index) 0)
+      (set-vector! (-> s5-0 local-sphere) -8192.0 0.0 0.0 40960.0)
+      (set-root-prim! gp-0 s5-0)
+      (let ((s4-0 (new 'process 'collide-shape-prim-mesh gp-0 (the-as uint 0) (the-as uint 0))))
+        (set! (-> s4-0 prim-core collide-as) (the-as uint 2048))
+        (set! (-> s4-0 collide-with) (the-as uint 16))
+        (set! (-> s4-0 prim-core action) (the-as uint 3))
+        (set! (-> s4-0 prim-core offense) 4)
+        (set! (-> s4-0 transform-index) 4)
+        (set-vector! (-> s4-0 local-sphere) 8192.0 0.0 0.0 28672.0)
+        (append-prim s5-0 s4-0)
+        )
+      )
+    (set! (-> gp-0 nav-radius) (* 0.75 (-> gp-0 root-prim local-sphere w)))
+    (backup-collide-with-as gp-0)
+    (set! (-> self root-override) gp-0)
+    gp-0
+    )
+  )
+
+(defstate pusher-idle (pusher)
+  :trans
+  (the-as (function none :behavior pusher) rider-trans)
+  :code
+  (behavior ()
+    (while #t
+      (let ((f0-0 -1.0))
+        (when (and *target* *camera*)
+          (let ((gp-0 (new 'stack-no-clear 'vector)))
+            (vector-! gp-0 (target-pos 0) (camera-pos))
+            (set! f0-0 (ray-capsule-intersect (-> self cyl) (camera-pos) gp-0))
```

Comment:
I think this checks to see if the camera is inside the pusher, and then retracts the pusher out of the way of the camera. Kind of cool

Reply:
I wonder if you were able to flip the camera around to face jak -- if they'd never get in your way.